src/measurements.ts

```diff
@@ -1,80 +1,183 @@
+    return sphere;
   }
 
   private calculateDistance() {
     if (this.points.length === 2) {
-      const result = {
-        x: Utils.roundOff(Math.sqrt(this.points[0].x*this.points[0].x + this.points[1].x*this.points[1].x)),
-        y: Utils.roundOff(Math.sqrt(this.points[0].y*this.points[0].y + this.points[1].y*this.points[1].y)),
-        z: Utils.roundOff(Math.sqrt(this.points[0].z*this.points[0].z + this.points[1].z*this.points[1].z)),
-        d: Utils.roundOff(this.points[0].distanceTo(this.points[1]))
-      }
-      this.updateMeasurementDisplay(`Distance = ${JSON.stringify(result)}`);
+      const distance = Utils.roundOff(this.points[0].distanceTo(this.points[1]));
+      this.distances.push(distance);
+      this.updateMeasurementDisplay();
     }
   }
 
-  private drawLine(scene: THREE.Scene) {
+  private drawLine() {
     const geometry = new THREE.BufferGeometry().setFromPoints(this.points);
-    this.line = new THREE.Line(geometry, new THREE.LineBasicMaterial({ color: 0xff0000 }));
-    scene.add(this.line);
+    const line = new THREE.Line(geometry, new THREE.LineBasicMaterial({ color: this.color }));
+    this.colors.push(this.color);
+    this.lines.push(line);
+    this.scene.add(line);
+  }
+
+  private generateRandomColor() {
+    return `#${Math.floor(Math.random() * 16777215).toString(16).padStart(6, '0')}`;
+  }
+
+  private resetPoints() {
+    this.points = [];
+    this.getRandomColor = true;
+  }
+
+  private convertDistance(distanceInMeters: number, unit: string): number {
+    switch (unit) {
+      case 'cm':
+        return distanceInMeters * 100;
+      case 'inch':
+        return distanceInMeters * 39.3701;
+      case 'foot':
+        return distanceInMeters * 3.28084;
+      case 'm':
+      default:
+        return distanceInMeters;
+    }
   }
 
-  private updateMeasurementDisplay(text: string) {
-    const measurementContainer = document.getElementById("measurementDiv");
-    if (measurementContainer) measurementContainer.innerText = text;
+  private updateMeasurementDisplay() {
+    const measurementContainer = document.querySelector("#table-body");
+    const unitSelect = document.getElementById("unit-select") as HTMLSelectElement;
+    if (!measurementContainer || !unitSelect) return;
+
+    const selectedUnit = unitSelect.value;
+
+    let rows = this.distances.map((distance, index) => {
+      const convertedDistance = this.convertDistance(distance, selectedUnit).toFixed(2);
+      return `
+        <tr>
+          <td><div style="width: 60px; height: 25px; background-color: ${this.colors[index]}; border: 1px solid black;"></div></td>
+          <td>${convertedDistance} ${selectedUnit}</td>
+          <td><button class="removeLine" data-index="${index}">Remove Line</button></td>
+        </tr>`;
+    }).join("");
+
+    if (this.distances.length > 0) {
+      rows += `
+        <tr>
+          <td></td>
+          <td></td>
+          <td><button id="clearButton">Reset All</button></td>
```

Comment:
```suggestion
          <td><button id="clearButton">Reset All</button></td>
```

I think the reset all button can be on the headers instead of adding a new row for that